quesma/queryparser/pancake_model.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"quesma/logger"
+	"quesma/model"
+)
+
+type pancakeAggregationTopLevel struct {
```

Comment:
It seems that `pancakeAggregationTopLevel` is subtype of pancakeAggregationLevel, why do we have distinction between them, should not we have just one type?

Reply:
From my understanding `pancakeAggregationTopLevel` holds a forest, and `pancakeAggregationLevel` represents a single tree.